docs/RFCS/range_leases.md

```diff
@@ -45,53 +45,68 @@ its replicas holding range leases at once?
+We introduce a new node liveness table at the beginning of the keyspace
 (not an actual SQL table; it will need to be accessed with lower-level
-APIs). This table is special in several respects: it is gossipped, and
+APIs). This table is special in several respects: it is gossiped, and
 leases within its keyspace (and all ranges that precede it, including
 meta1 and meta2) use the current, per-range lease mechanism to avoid
-circular dependencies. This table maps node IDs to an epoch counter
-and a lease expiration timestamp.
-
-The range lease is moved from a special raft command (which writes to
-a range-local, non-transactional range lease key) to a transactional
-range-local key (similar to the range descriptor). The range lease
-identifies the node that holds the lease and its epoch counter. It has
-a start timestamp but does not include an expiration time. The range
-lease record is always updated in a distributed transaction with the
-node lease record to ensure that the epoch counter is consistent and
-the start time is greater than the prior range lease holder’s node
-lease expiration (plus the maximum clock offset).
-
-Each node periodically performs a conditional put to its node lease to
-increase the expiration timestamp and ensure that the epoch has not
-changed. If the epoch does change, *all* of the range leases held by
-this node are revoked. A node *must not* propose any commands with a
-timestamp greater than the latest expiration timestamp it has written
-to the node lease table.
+circular dependencies. This table maps node IDs to an epoch counter,
+and an expiration timestamp.
+
+## Liveness table
+
+Column|     Description
+NodeID|     node identifier
+Epoch|      monotonically increasing liveness epoch
+Expiration| timestamp at which the liveness record expires
+
+Each node periodically heartbeats the liveness table, which is
+implemented as a conditional put which increases the expiration
+timestamp and ensures that the epoch has not changed. If the epoch
+does change, *all* of the range leases held by this node are
+revoked. A node can only execute commands (propose writes to Raft or
+serve reads) if it's the range leaseholder, the range lease epoch is
+equal to the node's liveness epoch, and the command timestamp is less
+than the node's liveness expiration minus the maximum clock offset.
+
+The node liveness table is gossiped by the range lease holder for the
+range which contains it. Gossip is used in order to minimize fanout
+and make distribution efficient. The best-effort nature of gossip is
+acceptable here because timely delivery of node liveness updates are
+not required for system correctness. Any node which fails to receive
+liveness updates will simply resort to a conditional put to increment
+a seemingly not-live node's liveness epoch. The conditional put will
+fail because the expected value is out of date and the correct liveness
+info is returned to the caller.
```

Comment:
This paragraph is better placed further down. Perhaps after the next two paragraphs